compiler/src/dmd/expression.d

```diff
@@ -249,18 +249,63 @@ bool isDotOpDispatch(Expression e)
 }
 
 /****************************************
- * Expand tuples.
- * Input:
- *      exps    aray of Expressions
- * Output:
- *      exps    rewritten in place
+ * Expand tuples in-place.
+ *
+ * Example:
+ *     When there's a call `f(10, pair: AliasSeq!(20, 30), single: 40))`, the input is:
+ *         exps =  [10, (20, 30), 40]
+ *         names = [null, "pair", "single"]);
+ *     The arrays will be modified to:
+ *         exps =  [10, 20, 30, 40]
+ *         names = [null, "pair", null, "single"]);
+ *
+ * Params:
+ *     exps  = array of Expressions
+ *     names = optional array of names corresponding to Expressions
  */
-extern (C++) void expandTuples(Expressions* exps)
+extern (C++) void expandTuples(Expressions* exps, Identifiers* names = null)
 {
     //printf("expandTuples()\n");
     if (exps is null)
         return;
 
+    if (names)
+    {
+        if (exps.length != names.length)
+        {
+            // Fixme: this branch should be removed when CallExp rewrites
+            // (UFCS/operator overloading) are fixed to take named arguments into account
+            names.setDim(exps.length);
+            foreach (i; 0..exps.length)
+                (*names)[i] = null;
```

Comment:
What are the situations where `names.length != exps.length`?  I assume from your comment that this is here to catch bugs in other places, however, why do you nullify the `names` array? Also, you're printing the elements in the next branch so you're putting yourself at risk of a segfault.

Reply:
The next branch won't be taken, because this branch does `names.setDim(exps.length);`. It nullifies them because `setDim` doesn't initialize new elements, and you don't want it to search a garbage identifier. I could just nullify the new elements, but it doesn't matter, the names aren't correct anyway. This is not a problem, because named arguments don't exist yet, so there's no code relying on it being correct. This whole branch is just scaffolding to make it not crash for various constructed `CallExp`s.

I don't know all situations where `names.length != exps.length`. One of them was where `a.fun(name: b)` gets rewritten to `fun(a, b)` without also prepending a `null` name, but I fixed that. There are a lot more, but I'll defer them to the PR that implements named function arguments.

Reply:
Ok, makes sense, thanks for the explanation.